psh/hd/hd.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * Phoenix-RTOS
+ *
+ * hd - prints file contents in hexadecimal and ascii representation
+ *
+ * Copyright 2023-2024 Phoenix Systems
+ * Author: Gerard Swiderski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <ctype.h>
+#include <fcntl.h>
+#include <getopt.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/minmax.h>
+#include <sys/stat.h>
+
+#include "../psh.h"
+
+#define LINE_WIDTH 16u
+
+
+static void psh_hd_info(void)
+{
+	printf("dumps file contents in hexadecimal and ascii representation");
+}
+
+
+static void psh_hd_usage(const char *name)
+{
+	printf("Usage: %s -s offset -n length [filename]\n", name);
+}
+
+
+static void psh_hexdump(const uint8_t *data, size_t ofs, size_t len)
+{
+	while (len != 0u) {
+		size_t i;
+		size_t toprint = min(len, LINE_WIDTH);
+
+		printf("%08zx ", (size_t)ofs);
+
+		for (i = 0u; i < toprint; i++) {
```

Comment:
Turbo nitpick - to avoid var reusing
```suggestion
		for (size_t i = 0u; i < toprint; i++) {
```

Reply:
Done.